### PR TITLE
Add more gVCF support

### PR DESCRIPTION
**Pull request by cmdcolin**
This PR attempts to add more support for GVCF, breaking from #847 to allow that PR to be more single topic

Most gVCF files consists of blocks of "<NON_REF>" calls that span a range (e.g. it uses the END= info field).

Some special cases

1) escapeHTML is done to make the <NON_REF> type things printable in popup
2) reference_allele is ignored when there is END field because it appears this just takes on a junk value of a single dna letter
3) alternative_allele added even for these symbolic alleles, note that they can take the form of a list e.g. G,<NON_REF>. For this, note that the check for if(alt) that was added in #847 now goes away

Several sample data test files are added, including gatk haplotype caller run in gvcf mode run over the volvox-sorted bam file :)

### CODE REVIEW

File: src/JBrowse/View/DetailsMixin.js [hunk 1]
```diff
@@ -187,8 +189,7 @@ return declare( null, {
                 return keys.length;
             }
         }
-
-        domConstruct.create('div', { className: 'value '+class_, innerHTML: val }, parent );
+        domConstruct.create('div', { className: 'value '+class_, innerHTML: dojoxHtmlEntities.encode(''+val) }, parent );
```
[the remark this answers is not in the document]
Comment (cmdcolin) [reply src/JBrowse/View/DetailsMixin.js]: I guess the solution could just be to remove the <> in the alt field in the first place, but it sort of helps to signal to the end user what it came from

[the remark this answers is not in the document]
Comment (rbuels) [reply src/JBrowse/View/DetailsMixin.js]: I reverted the html-entitizing and the display didn't look any different to me, but maybe I'm looking at the wrong thing.

anyway, a little-known feature of the detail pages is: if a thing (any data type) has a `toHTML` method on it, the detail page will call it and display that HTML.  So, you could add toHTML methods to some things in VariantDetailMixin, maybe?

**Comment by rbuels** [comment]
I reverted the html-entitizing and the display didn't look any different to me, but maybe I'm looking at the wrong thing.

anyway, a little-known feature of the detailmixin is: if a thing (any data type) has a toHTML method on it, the detail page will call it and display that HTML. So, you could add toHTML methods to some things in VariantDetailMixin, maybe?

**Comment by cmdcolin** [comment]
The <*> actually doesn't cause problems but other symbolic alleles do including <DEL:ME> in the volvox test data and <NON_REF> from gvcf and other things. Note that on the current live jbrowse.org instance, it just displays 1|1 in the genotype field on the <DEL:ME> in the genotype table, but with this PR it tries to pull the actual alt allele text (as it should IMO) into the genotype table

Here's a screenshot displaying the issue


The encoding entities nicely fixed both the the alternative allele field and the genotype table but possibly could look for another workaround


screenshot-localhost-2018 03 27-11-42-01

**Comment by cmdcolin** [comment]
Looks like the reason that the dgrid was fixed also was that renderDetailValue got called on the data columns of the dgrid as well as those regular detail value fields

**Comment by cmdcolin** [comment]
I fixed the rendering on the genotype table but didn't get the alternative_alleles field to render (couldn't get the toHTML function working)

**Comment by cmdcolin** [comment]
Code should now escape only the alternative_alleles field in popup as requested :)

**Comment by rbuels** [comment]
awesome, nice job!

**Pull request merged**